compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirDeclarationsResolveTransformer.kt

```diff
@@ -41,6 +42,13 @@ import org.jetbrains.kotlin.name.Name
 open class FirDeclarationsResolveTransformer(transformer: FirBodyResolveTransformer) : FirPartialBodyResolveTransformer(transformer) {
     private var containingClass: FirRegularClass? = null
 
+    private fun <T> withFirArrayOfCallTransformer(block: () -> T): T {
```

Comment:
Please make this function `inline` and put `transformer.expressionsTransformer.enableArrayOfCallTransformation = false` into `finally` block so it will be called even if there was a `return` inside `block`

Reply:
Thank you for the suggestions. Done.